libraries/Microsoft.Bot.Builder.LanguageGeneration/Template.cs

```diff
@@ -20,140 +17,62 @@ public class Template
         /// <summary>
         /// Initializes a new instance of the <see cref="Template"/> class.
         /// </summary>
-        /// <param name="parseTree">The parse tree of this template.</param>
-        /// <param name="lgfileContent">lg file content.</param>
-        /// <param name="source">Source of this template.</param>
-        internal Template(LGFileParser.TemplateDefinitionContext parseTree, string lgfileContent, string source = "")
+        /// <param name="templateName">Template name without parameters.</param>
+        /// <param name="parameters">Parameter list.</param>
+        /// <param name="templateBody">Template content.</param>
+        /// <param name="sourceRange">souerce range of template.</param>
+        internal Template(
+            string templateName,
+            List<string> parameters,
+            string templateBody,
+            SourceRange sourceRange)
         {
-            ParseTree = parseTree;
-            Source = source;
-
-            Name = ExtractName();
-            Parameters = ExtractParameters();
-            Body = ExtractBody(lgfileContent);
+            this.Name = templateName ?? string.Empty;
+            this.Parameters = parameters ?? new List<string>();
+            this.Body = templateBody ?? string.Empty;
+            this.SourceRange = sourceRange;
         }
 
         /// <summary>
-        /// Gets name of the template, what's followed by '#' in a LG file.
+        /// Gets or sets source range.
         /// </summary>
         /// <value>
-        /// Name of the template, what's followed by '#' in a LG file.
+        /// start line of the tempalte in LG file.
         /// </value>
-        public string Name { get; }
+        public SourceRange SourceRange { get; set; }
 
         /// <summary>
-        /// Gets parameter list of this template.
+        /// Gets or sets name of the template, what's followed by '#' in a LG file.
```

Comment:
>what's followed [](start = 47, length = 15)

which follows

Reply:
"which follows '#' in an LG file"

---
In reply to: [409984499](https://github.com/microsoft/botbuilder-dotnet/pull/3731#discussion_r409984499) [](ancestors = 409984499)